model/at_mentions.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
+
+import (
+	"regexp"
+	"strings"
+)
+
+var atMentionRegexp = regexp.MustCompile(`\B@[[:alnum:]][[:alnum:]\.\-_]*`)
+
+const usernameSpecialChars = ".-_"
+
+// PossibleAtMentions returns all substrings in message that look like valid @
+// mentions.
+func PossibleAtMentions(message string) []string {
+	var names []string
+
+	if strings.Contains(message, "@") {
```

Comment:
nit: suggest inverting and returning early to reduce indentation here

Reply:
Done in https://github.com/mattermost/mattermost-server/pull/13762/commits/552c67e6d459e98b646fd149c792a2bb0eb7486d.